### ISSUE TITLE
iOS 8 support

### ISSUE THREAD

**Issue by RamwiseMatt** [SwiftKickMobile/SwiftMessages#15]
As I understand from the podspec file, the minimum required OS is iOS 8. However, the use of UIStackView in the codebase causes a crash on iOS 8, while working fine on iOS 9. Should I assume the minimum requirement is actually iOS 9?

**Comment by wtmoose** [comment]
No. The code does not make any references to `UIStackView`, so the library works just fine in iOS 8. You just can't use the stack view-based layouts.

The `MessageViewIOS8` layout is provided for iOS 8 use specifically. If it doesn't do exactly what you need, you can copy `MessageViewIOS8.nib` into your project and change it. I will consider adding more iOS 8-friendly layouts if `MessageViewIOS8.nib` doesn't work for you.

**Comment by wtmoose** [comment]
No response from @RamwiseMatt. Closing.

**Comment by dogo** [comment]
Hi @wtmoose 

In fact the crash is occurring with iOS 8.x. My log running with iOS 8.1

```
2017-01-04 02:02:07.677 SWDestiny Trades[27080:203721] *** Terminating app due to uncaught exception 'NSInvalidUnarchiveOperationException', reason: 'Could not instantiate class named UIStackView'
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001117b8f35 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x0000000111453bb7 objc_exception_throw + 45
	2   CoreFoundation                      0x00000001117b8e6d +[NSException raise:format:] + 205
	3   UIKit                               0x00000001122b753d UINibDecoderDecodeObjectForValue + 312
	4   UIKit                               0x00000001122b7895 UINibDecoderDecodeObjectForValue + 1168
	5   UIKit                               0x00000001122b73fc -[UINibDecoder decodeObjectForKey:] + 276
	6   UIKit                               0x0000000111ee9ca6 -[UIView initWithCoder:] + 855
	7   UIKit                               0x00000001122b76c6 UINibDecoderDecodeObjectForValue + 705
	8   UIKit                               0x00000001122b7895 UINibDecoderDecodeObjectForValue + 1168
	9   UIKit                               0x00000001122b73fc -[UINibDecoder decodeObjectForKey:] + 276
	10  UIKit                               0x0000000111ee9ca6 -[UIView initWithCoder:] + 855
	11  SwiftMessages                       0x0000000110dfea39 _TFC13SwiftMessages8BaseViewcfT5coderCSo7NSCoder_GSqS0__ + 233
	12  SwiftMessages                       0x0000000110e03ebc _TFC13SwiftMessages11MessageViewcfT5coderCSo7NSCoder_GSqS0__ + 204
	13  SwiftMessages                       0x0000000110e03f5d _TToFC13SwiftMessages11MessageViewcfT5coderCSo7NSCoder_GSqS0__ + 45
	14  UIKit                               0x00000001121529dd -[UIClassSwapper initWithCoder:] + 205
	15  UIKit                               0x00000001122b76c6 UINibDecoderDecodeObjectForValue + 705
	16  UIKit                               0x00000001122b73fc -[UINibDecoder decodeObjectForKey:] + 276
	17  UIKit                               0x00000001121525aa -[UIRuntimeConnection initWithCoder:] + 109
	18  UIKit                               0x00000001122b76c6 UINibDecoderDecodeObjectForValue + 705
	19  UIKit                               0x00000001122b7895 UINibDecoderDecodeObjectForValue + 1168
	20  UIKit                               0x00000001122b73fc -[UINibDecoder decodeObjectForKey:] + 276
	21  UIKit                               0x0000000112151a79 -[UINib instantiateWithOwner:options:] + 990
	22  UIKit                               0x0000000112153722 -[NSBundle(UINSBundleAdditions) loadNibNamed:owner:options:] + 160
	23  SwiftMessages                       0x0000000110e1f8ab _TZFC13SwiftMessages13SwiftMessagesP33_AB28F0CA9BFF0B16E059C4C20EF9AFAA19internalViewFromNibuRxCSo6UIViewrfzT5namedSS6bundleGSqCSo6Bundle_10filesOwnerPs9AnyObject__x + 1355
	24  SwiftMessages                       0x0000000110e1fd9c _TZFC13SwiftMessages13SwiftMessages11viewFromNibuRxCSo6UIViewrfzT5namedSS10filesOwnerPs9AnyObject__x + 156
	25  SwiftMessages                       0x0000000110e04b6f _TZFC13SwiftMessages11MessageView11viewFromNibuRxS0_rfT6layoutOS0_6Layout10filesOwnerPs9AnyObject__x + 143
	26  SWDestiny Trades                    0x000000010fa24c5d _TFC16SWDestiny_Trades21AddCardViewControllerP33_CABF6C98A2FBF9228C9FD15D9AB64AEA18showSuccessMessagefT4cardCS_7CardDTO_T_ + 61
	27  SWDestiny Trades                    0x000000010fa2468f _TFFC16SWDestiny_Trades21AddCardViewControllerP33_CABF6C98A2FBF9228C9FD15D9AB64AEA6insertFT2atV10Foundation9IndexPath_T_U_FT_T_ + 847
	28  RealmSwift                          0x0000000110cdbf47 _TFC10RealmSwift5Realm5writefzFzT_T_T_ + 71
	29  SWDestiny Trades                    0x000000010fa242d8 _TFC16SWDestiny_Trades21AddCardViewControllerP33_CABF6C98A2FBF9228C9FD15D9AB64AEA6insertfT2atV10Foundation9IndexPath_T_ + 200
	30  SWDestiny Trades                    0x000000010fa2396c _TFC16SWDestiny_Trades21AddCardViewController9tableViewfTCSo11UITableView14didSelectRowAtV10Foundation9IndexPath_T_ + 60
	31  SWDestiny Trades                    0x000000010fa23aa7 _TToFC16SWDestiny_Trades21AddCardViewController9tableViewfTCSo11UITableView14didSelectRowAtV10Foundation9IndexPath_T_ + 87
	32  UIKit                               0x0000000111f74393 -[UITableView _selectRowAtIndexPath:animated:scrollPosition:notifyDelegate:] + 1293
	33  UIKit                               0x0000000111f744d4 -[UITableView _userSelectRowAtPendingSelectionIndexPath:] + 219
	34  UIKit                               0x0000000111eaf331 _applyBlockToCFArrayCopiedToStack + 314
	35  UIKit                               0x0000000111eaf1ab _afterCACommitHandler + 516
	36  CoreFoundation                      0x00000001116eddc7 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
	37  CoreFoundation                      0x00000001116edd20 __CFRunLoopDoObservers + 368
	38  CoreFoundation                      0x00000001116e3b53 __CFRunLoopRun + 1123
	39  CoreFoundation                      0x00000001116e3486 CFRunLoopRunSpecific + 470
	40  GraphicsServices                    0x0000000114cee9f0 GSEventRunModal + 161
	41  UIKit                               0x0000000111e8c420 UIApplicationMain + 1282
	42  SWDestiny Trades                    0x000000010fa17a5f main + 111
	43  libdyld.dylib                       0x0000000113b66145 start + 1
	44  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) ```

**Comment by dogo** [comment]
ohhh never mind , I should explicitly set .MessageViewIOS8.

Thanks

**Comment by wtmoose** [comment]
Yep. That's my way of nudging you to drop iOS 8 :)

**Comment by dogo** [comment]
```available``` do the job :P

``` swift
let success: MessageView
if #available(iOS 9.0, *) {
      success = MessageView.viewFromNib(layout: .CardView)
} else {
      success = MessageView.viewFromNib(layout: .MessageViewIOS8)
}
```
